scipy/sparse/tests/test_array_api.py

```diff
@@ -343,7 +347,11 @@ def test_spilu():
         [4, 3, 2, 1],
     ])
     LU = spla.spilu(X)
-    npt.assert_allclose(LU.solve(np.array([1, 2, 3, 4])), [1, 0, 0, 0])
+    npt.assert_allclose(
+        LU.solve(np.array([1, 2, 3, 4])),
+        np.asarray([1, 0, 0, 0], dtype=np.float64),
+        rtol=1e-14, atol=3e-16
+    )
```

Comment:
@stefanv @perimosocordiae I merged this PR to move things along for i.e., eventual 64-bit SuperLU support, patch upstreaming, etc. The testing shims don't look too controversial to me, but if there is something that worries you can you ping me? It was just the new coercion that made me slightly nervous as non-expert reviewer on sparse linalg.

Reply:
IIRC this was a matter of the result from `LU.solve` being on the order of 1e-17 rather than exactly zero, and the default `atol=0` not working for that. It's been a few weeks, but IIRC this is business as usual with minor differences in floating-point results.